com.unity.netcode.gameobjects/Runtime/Components/Interpolator/BufferedLinearInterpolator.cs

```diff
@@ -350,23 +428,63 @@ private void TryConsumeFromBuffer(double renderTime, float minDeltaTime, float m
         /// <param name="tickLatencyAsTime">The tick latency in relative local time.</param>
         /// <param name="minDeltaTime">The minimum time delta between the current and target value.</param>
         /// <param name="maxDeltaTime">The maximum time delta between the current and target value.</param>
+        /// <param name="isLerpAndExtrapolate">Determines whether to use smooth dampening or extrapolation.</param>
         /// <returns>The newly interpolated value of type 'T'</returns>
-        internal T Update(float deltaTime, double tickLatencyAsTime, float minDeltaTime, float maxDeltaTime)
+        internal T Update(float deltaTime, double tickLatencyAsTime, double minDeltaTime, double maxDeltaTime, bool isLerpAndExtrapolate)
         {
-            TryConsumeFromBuffer(tickLatencyAsTime, minDeltaTime, maxDeltaTime);
-            // Only interpolate when there is a start and end point and we have not already reached the end value
+            TryConsumeFromBuffer(tickLatencyAsTime, minDeltaTime, maxDeltaTime, isLerpAndExtrapolate);
+            // Only begin interpolation when there is a start and end point
             if (InterpolateState.Target.HasValue)
             {
-                InterpolateState.AddDeltaTime(deltaTime);
-
-                // Smooth dampen our current time
-                var current = SmoothDamp(InterpolateState.CurrentValue, InterpolateState.Target.Value.Item, ref m_RateOfChange, InterpolateState.TimeToTargetValue, InterpolateState.DeltaTime);
-                // Smooth dampen a predicted time based on our average delta time 
-                var predict = SmoothDamp(InterpolateState.CurrentValue, InterpolateState.Target.Value.Item, ref m_PredictedRateOfChange, InterpolateState.TimeToTargetValue, InterpolateState.DeltaTime + (InterpolateState.AverageDeltaTime * 2));
-                // Lerp between the current and predicted.
-                // Note: Since smooth dampening cannot over shoot, both current and predict will eventually become the same or will be very close to the same.
-                // Upon stopping motion, the final resing value should be a very close aproximation of the authority side.
-                InterpolateState.CurrentValue = Interpolate(current, predict, deltaTime);
+                // As long as the target hasn't been reached, interpolate or smooth dampen.
+                if (!InterpolateState.TargetReached)
+                {
+                    // Increases the time delta relative to the time to target.
+                    // Also calculates the LerpT and LerpTPredicted values.
+                    InterpolateState.AddDeltaTime(deltaTime);
+                    var targetValue = InterpolateState.CurrentValue;
+                    // SmoothDampen or LerpExtrapolateBlend
+                    if (!isLerpAndExtrapolate)
```

Comment:
Where this function is called `isLerpAndExtrapolate` is being set as `PositionInterpolationType == InterpolationTypes.LerpExtrapolateBlend`. This code comment doesn't seem to match.

Reply:
It evaluates to a Boolean where if using `InterpolationTypes.LerpExtrapolateBlend` it Lerps then Lerps unclamped (extrapolates) where if it is not then it does the two pass smooth dampening. It is sort of inverted logic...
Use the smooth dampening if we are not doing `LerpExtrapolateBlend` is what that turns out to be logically.

Reply:
I missed that there were two `Update` functions! I follow it now